File: Mavim-apps/Mavim.frontend/projects/mavim-app/src/shared/field/mappers/multidecimalfield.mapper.ts
import { Field } from '../models/field.model';
import { MultiDecimalField } from '../models/api';
import { GenericDecimalFieldMapper } from './abstract';
import { mapToApiFieldType, mapToFieldType } from './fieldvaluetype';

export class MultiDecimalFieldMapper extends GenericDecimalFieldMapper<MultiDecimalField> {
	public get getUrlTypeName(): string {
		return 'multidecimal';
	}

	public getMappedField(field: MultiDecimalField): Field {
		return this.mapField(field);
	}

	public getMappedApiField(field: Field): MultiDecimalField {
		return this.mapApiField(field);
	}

	protected mapApiField(field: Field): MultiDecimalField {
		const data =
			field.data && field.data.length > 0
				? field.data.map((d) => this.mapToDecimal(d))
				: [];
		return {
			...field,
			fieldValueType: mapToApiFieldType(field.fieldValueType),
			data
		} as MultiDecimalField;
	}

	protected mapField(field: MultiDecimalField): Field {
		const data =
			field.data && field.data.length > 0
				? field.data.map((d) => this.mapToString(d))
				: [];

		return {
			...field,
			fieldValueType: mapToFieldType(field.fieldValueType),
			data
		} as Field;
	}
}

File: Mavim-services/Topic/Mavim.Manager.Api.Topic.Repository/v1/Mappers/MultiDecimalFieldMapper.cs
using Mavim.Manager.Api.Topic.Repository.Interfaces.v1.enums;
using Mavim.Manager.Api.Topic.Repository.Interfaces.v1.Fields;
using Mavim.Manager.Api.Topic.Repository.v1.Fields;
using Mavim.Manager.Api.Topic.Repository.v1.Mappers.Abstract;
using System;
using System.Linq;

namespace Mavim.Manager.Api.Topic.Repository.v1.Mappers
{
    internal class MultiDecimalFieldMapper : GenericDecimalFieldMapper<Model.IMultiValueSimpleField, IMultiDecimalField>
    {
        /// <summary>
        /// Gets the generic mapped multi decimal service field.
        /// </summary>
        /// <param name="field">The field.</param>
        /// <returns></returns>
        protected override IMultiDecimalField GetGenericMappedServiceField(Model.IMultiValueSimpleField field)
        {
            if (field == null) throw new ArgumentNullException(nameof(field));

            return new MultiDecimalField
            {
                FieldId = field.FieldDefinition.ID,
                FieldName = field.FieldDefinition.Name,
                SetOrder = field.FieldSetDefinition.OrderNumber,
                Order = field.OrderNumber,
                FieldSetId = field.FieldSetDefinition.ID,
                SetName = field.FieldSetDefinition?.Name ?? string.Empty,
                FieldValues = field.FieldValues?.Select(i => i?.ToString()),
                FieldValueType = FieldType.MultiDecimal,
                Required = field.FieldDefinition.Required,
                Readonly = field.ReadOnly
            };
        }

        /// <summary>
        /// Gets the generic mapped multi decimal repo field.
        /// </summary>
        /// <param name="field">The field.</param>
        /// <param name="simpleField">The simple field.</param>
        /// <returns></returns>
        protected override object[] GetGenericMappedRepoField(IMultiDecimalField field, Model.ISimpleField simpleField = null)
        {
            if (field == null) return new object[] { null };

            return field.FieldValues == null ? new object[0] : field.FieldValues.ToArray<object>();
        }
    }
}
